code/modules/admin/verbs/admin_verb_datum/_admin_verb.dm

```diff
@@ -0,0 +1,134 @@
+	holder = verb_datum
+	new /mob/admin_verb_holder/proc/_wrap(src, holder.verb_name, holder.verb_desc)
+
+/mob/admin_verb_holder/proc/_wrap()
+	// We use group to act as a list of clients allowed to access this verb
+	set src in usr.group
+
+	// But we still double check rights
+	if(check_rights_for(usr.client, holder.permission_required))
+		SSblackbox.record_feedback("tally", "admin_verb_datum", 1, holder.verb_name) // remember to record feedback before invocation
+		holder.invoke(usr.client, holder.get_arguments(usr.client))
+
+/**
+ * The base admin verb datum.
+ * Essentially just a more readable wrapper for admin verbs over dumping them in a giant ass list.
+ */
+/datum/admin_verb_datum
+	/// The name that will appear in the verb panel
+	var/verb_name = "Default Admin Verb"
+	/// An optional, CANNOT BE NULL, description to appear when you hover over a verb in the panel.
+	var/verb_desc = ""
+	/// The category of this verb, needs to be set to something
+	var/verb_category = "Admin"
+
+	/// The permissions required to both see and invoke this datum verb
+	var/permission_required
+	/// The abstract of the verb datum, to prevent creating abstract types
+	var/abstract = /datum/admin_verb_datum
+
+	/// The verb holder for this datum, ensuring admins retain verb bar usage
+	VAR_PRIVATE/mob/admin_verb_holder/verb_holder
+	/// An assosciative list of client -> callbacks[] to allow for mob tracking to update verb grouping
+	VAR_PRIVATE/list/client_to_callbacks
+
+GENERAL_PROTECT_DATUM(/datum/admin_verb_datum)
+
+/datum/admin_verb_datum/New()
+	SHOULD_NOT_OVERRIDE(TRUE)
+	verb_holder = new(null, src)
+
+/datum/admin_verb_datum/Topic(href, list/href_list)
+	..()
+	if(!usr.client?.holder?.CheckAdminHref(href, href_list))
+		return
+	if(!check_rights_for(usr.client, permission_required))
+		return
+
+	if(href_list["invoke"])
+		SSblackbox.record_feedback("tally", "admin_verb_datum", 1, verb_name) // remember to record feedback before invocation
+		invoke(usr.client, get_arguments(usr.client))
+
+// The following procs only exist to support verb bar usage and can be removed when that is no longer required.
+
+/datum/admin_verb_datum/proc/on_mob_login(mob/logged_in)
+	SHOULD_NOT_OVERRIDE(TRUE)
+	logged_in.group |= verb_holder
+
+/datum/admin_verb_datum/proc/on_mob_logout(mob/logged_out)
+	SHOULD_NOT_OVERRIDE(TRUE)
+	logged_out.group -= verb_holder
+
+/datum/admin_verb_datum/proc/assosciate(client/admin)
+	SHOULD_NOT_OVERRIDE(TRUE)
+	if(!check_rights_for(admin, permission_required))
+		return
+
+	var/datum/callback/on_login = CALLBACK(src, PROC_REF(on_mob_login))
+	var/datum/callback/on_logout = CALLBACK(src, PROC_REF(on_mob_logout))
+	LAZYSET(client_to_callbacks, admin.ckey, list(on_login, on_logout))
+
+	var/datum/player_details/player_details = admin.player_details
+
+	player_details.post_login_callbacks += on_login
+	player_details.post_logout_callbacks += on_logout
+
+	// If this is being called in client/New they will not actually have a mob yet!
```

Comment:
This can be fixed by moving such calls to after the `..()` in client/New. as `..()` is what sets the mob.